FCWTAPI/PlottingUtils.cs

```diff
@@ -339,17 +420,17 @@ public static PlotModel GenerateXYPlotCWT(double[,] data, int[] rowIndices, doub
         /// Exports generated plots to PDF files
         /// </summary>
         /// <param name="plotModel">Plot to export</param>
-        /// <param name="fileName">Filename to export the plot to, must contain .pdf extension</param>
+        /// <param name="filePath">File path to export the plot to, must contain .pdf extension</param>
         /// <param name="plotWidth">Width of the plot</param>
         /// <param name="plotHeight">Height of the plot</param>
         /// <exception cref="ArgumentException"></exception>
-        public static void ExportPlotPDF(PlotModel plotModel, string fileName, int plotWidth = 700, int plotHeight = 600)
+        public static void ExportPlotPDF(PlotModel plotModel, string filePath, int plotWidth = 700, int plotHeight = 600)
         {
-            if (fileName.Substring(fileName.Length - 4, 4) != ".pdf")
+            if (Path.GetExtension(filePath) != ".pdf")
```

Comment:
Nice exception handling, but since you're already detecting if the pdf extension is present, you should just add code to append .pdf on the end of the file name if it isn't present. Also, in the future, there may be other options besides pdf format, so got to keep it flexible.

Reply:
I'm going to leave this for now, just because fixing it will require some more extensive editing of the PlottingUtils unit tests, but I'll make a note to fix this over the next day or so while I finish cleaning up the code and fixing small issues.